### ISSUE TITLE
ASP.NET Core Applications will not run without Datadog.Trace.ClrProfiler.Managed package installed

### ISSUE THREAD

**Issue by mclark1129** [DataDog/dd-trace-dotnet#429]
**Describe the bug**
When DD APM is installed and automatically instrumenting IIS application, ASP.NET core apps will fail to run if the `Datadog.Trace.ClrProfiler.Managed` is not installed alongside it, or if the version of the package does not match the version of the tracer.

**To Reproduce**
1. Install the .NET Tracer on a machine running IIS
2. Deploy an ASP.NET Core application that does not have `Datadog.Trace.ClrProfiler.Managed` installed

**Expected behavior**
The application functions normally, likely without reporting tracing to DD APM

**Actual Behavior** 
The application throws a `FileNotFoundException`

```
System.IO.FileNotFoundException: Could not load file or assembly 'Datadog.Trace.ClrProfiler.Managed, Version=1.4.0.0, Culture=neutral, PublicKeyToken=def86d061d0d2eeb'. The system cannot find the file specified.
File name: 'Datadog.Trace.ClrProfiler.Managed, Version=1.4.0.0, Culture=neutral, PublicKeyToken=def86d061d0d2eeb'
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Server.IISIntegration.IISMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Builder.Extensions.UsePathBaseMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
```

**Runtime environment (please complete the following information):**
- Instrumentation mode: [e.g. automatic with msi installer or manual with NuGet package]
- Tracer version: 1.4.0.0
- OS: Windows Server 2012 R2 Standard
- CLR: .NET Core 2.0+ (Can reproduce on 2.0 and 2.2)

**Comment by colin-higgins** [comment]
Thank you for the bug report. We are looking into this.

**Comment by colin-higgins** [comment]
Hi @mclark1129 I just wanted to let you know that this is in progress with https://github.com/DataDog/dd-trace-dotnet/pull/442

**Comment by colin-higgins** [comment]
Hey @mclark1129, would you be willing to install the latest version of the tracer and check to see if this issue remains?
https://github.com/DataDog/dd-trace-dotnet/releases/tag/v1.6.1

**Comment by lucaspimentel** [comment]
This should be fixed on Windows with 1.6.1, and will be fixed on Linux with the upcoming 1.6.2. Let's keep the issue open until it's confirmed fixed on both platforms.

**Comment by lucaspimentel** [comment]
Update: we are releasing 1.6.2 with a fix for another issue, so _this_ fix on Linux is now scheduled for 1.6.3.

**Comment by mclark1129** [comment]
Thanks @colin-higgins and @lucaspimentel, I was OOO last week but I will make an effort to re-test this on one of our development environments this week.

**Comment by zacharycmontoya** [comment]
We've released version [1.7.0](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v1.7.0) which contains the Linux set of changes for the issue. I'll close this issue for now but if we see this behavior again we'll be happy to revisit the issue.
